### ISSUE TITLE
Cannot create subscription with GCP provider, payment_method_id is required

### ISSUE THREAD

**Issue by romanlv** [RedisLabs/terraform-provider-rediscloud#61]
### Terraform Version
Terraform v0.13.5

### Affected Resource(s)
Please list the resources as a list, for example:
- rediscloud_subscription

### Debug Output
https://gist.github.com/romanlv/fbed36587d45d1625f7d11600d80923d

### Expected Behavior
Should be able to create subscription and db without providing payment method

### Actual Behavior

> Error: strconv.Atoi: parsing "": invalid syntax

I can see that documentation has been updated recently to make `payment_method_id` optional, but actual code uses it as required input

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

```
resource "random_password" "password" {
  length = 20
  upper = true
  lower = true
  number = true
  special = false
}

resource "rediscloud_subscription" "example" {

  name = "example"
  // payment_method_id = "0" // data.rediscloud_payment_method.card.id
  memory_storage = "ram"
  persistent_storage_encryption = false

  cloud_provider {
    cloud_account_id = local.rediscloud_account_id
    provider = "GCP"

    region {
      region = "eu-west-1"
      networking_deployment_cidr = "10.0.0.0/24"
      preferred_availability_zones = ["eu-west-1a"]
    }
  }

  database {
    name = "tf-example-database"
    protocol = "redis"
    memory_limit_in_gb = 1
    data_persistence = "none"
    throughput_measurement_by = "operations-per-second"
    throughput_measurement_value = 10000
    password = random_password.password.result

    alert {
      name = "dataset-size"
      value = 40
    }
  }
}
```

**Comment by romanlv** [comment]
I tried to comment out payment_method_id in provider code, but started getting 

```Error: 400 BAD_REQUEST - SUBSCIPTION_GCP_ALLOW_ONLY_INTERNAL: GCP subscription is supported only with a Redis Labs internal resources``` 

Possible that api client needs to be updated too. 

Please let me know if it's an easy fix to make, or suggested workaround

**Comment by Aviad-Abutbul** [comment]
Hi @romanlv. Thanks for reporting this. We actually also noticed it and we already have a fix (see #62 and #56). This will be pushed into our next release.

**Comment by romanlv** [comment]
Thank you, I can confirm that it works with latest code on `main`.
